### ISSUE TITLE
DRM use flag fails without wayland support

### ISSUE THREAD

**Issue by jblandrum** [rafspiny/enlightenment-live#67]
[ebuild   R   *] dev-libs/efl-9999::gambit  USE="X avahi bmp connman eet fbcon fontconfig gif glib harfbuzz hyphen ibus ico jpeg2k libuv nls opengl pdf physics postscript ppm psd pulseaudio scim sdl sound ssl svg tga tiff v4l vlc vnc webp
xcf xim xine xpm xpresent -dds -debug -doc -drm* -egl -eo -example% -fribidi -gles -gnutls -gstreamer -json% -luajit -pixman -raw -systemd -test% -tslib -unwind -wayland (-libressl%) (-neon%) (-valgrind%)

Build works with this useflag set but if you turn on the DRM flag build breaks trying to #include <wayland-client.h> from part of the evas. 

Also with these useflags (although luagit and gstreamer are not turned on) I had to emerge --oneshot gstreamer gst-plugin-base luajit
in order to get the build to complete

**Comment by rafspiny** [comment]
Hi @jblandrum,

this is partially being handle by #66. There was a problem with the use of the `gstreamer` use flag.
I still need to look into the one for DRM/wayland issue.

If you can pastebin here or on IRC the full emerge output, it would be helpful.

**Comment by rafspiny** [comment]
Update: after a chat on IRC with raster it turned out that the DRM option on meson is actually dependant on wayland. If there is a need for DRM and wayland to be decoupled, this should be escalated to the EFL project cause the ebuild is actually getting that dependency right.

I am gonna close this issue after #66 is merged

**Comment by jblandrum** [comment]
I did get it to build by removing drm if that's the case specifying the drm flag should force the wayland flag on.
It doesn't look like the #66 ticket force drm to require the wayland use flag

I was having another issue with line 
https://github.com/rafspiny/enlightenment-live/blob/db8e697c20f8573cc7a3737f6664cab2265bc88d/dev-libs/efl/efl-9999.ebuild#L173
I was getting error '' is not a valid option in "libvlc, gstreamer, gstreamer1, xine"
And this was due to the comma placement at the end of gstreamer
It was choking on emerge outputting the trailing comma
-Demotion-loaders-disabler=libvlc,gstreamer,

so I change the line to what would be 
`-Demotion-loaders-disabler=libvlc,gstreamer$(usex gstreamer '' ',gstreamer1')$(usex xine '' ',xine')`
in the new

**Comment by rafspiny** [comment]
Good catch. The comma must be inside indeed!

**Comment by rafspiny** [comment]
@jblandrum  #68 address the two problems: DRM and gstreamer.

**Comment by rafspiny** [comment]
Closing this one. Feel free to re-open it if needed.
